src/Microsoft.OData.Core/JsonLight/ODataJsonLightSerializer.cs

```diff
@@ -294,34 +296,11 @@ internal Task WritePayloadEndAsync()
                 contextUrlInfo = contextUrlInfoGen();
             }
 
-            if (contextUrlInfo != null && contextUrlInfo.IsHiddenBy(parentContextUrlInfo))
-            {
-                return null;
-            }
-
-            Uri contextUri = ContextUriBuilder.BuildContextUri(payloadKind, contextUrlInfo);
-
-            if (contextUri != null)
-            {
-                if (string.IsNullOrEmpty(propertyName))
-                {
-                    await this.AsynchronousODataAnnotationWriter.WriteInstanceAnnotationNameAsync(ODataAnnotationNames.ODataContext)
-                        .ConfigureAwait(false);
-                }
-                else
-                {
-                    await this.AsynchronousODataAnnotationWriter.WritePropertyAnnotationNameAsync(propertyName, ODataAnnotationNames.ODataContext)
-                        .ConfigureAwait(false);
-                }
-
-                await this.AsynchronousJsonWriter.WritePrimitiveValueAsync(contextUri.IsAbsoluteUri ? contextUri.AbsoluteUri : contextUri.OriginalString)
-                    .ConfigureAwait(false);
-                this.allowRelativeUri = true;
-
-                return contextUrlInfo;
-            }
-
-            return null;
+            return await WriteContextUriPropertyImplementationAsync(

```

Comment:
Curious why this was not made a local method.

Reply:
Aah I see, you made multiple generic versions of this method to accommodate multiple args for the callback.